src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -141,6 +123,48 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
             }
         }
 
+        private static bool IsValidExpression(SyntaxNode expression, ISyntaxFactsService syntaxFacts)
+        {
+            // Need to special case for expressions that are contained within a parameter
+            // because it is technically "contained" within a method, but an expression in a parameter does not make
+            // sense to introduce.
+            var parameterNode = expression.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsParameter);
```

Comment:
This is now walking up the tree multiple times. Can we just do it once and return on the first problematic node?